sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatchResponse.java

```diff
@@ -0,0 +1,219 @@
+     * @param subStatusCode the response sub-status code.
+     * @param errorMessage an error message or {@code null}.
+     * @param responseHeaders the response http headers
+     * @param cosmosDiagnostics the diagnostic
+     * @param operations a {@link List list} of {@link ItemBatchOperation batch operations}.
+     */
+    TransactionalBatchResponse(
+        final int responseStatus,
+        final Integer subStatusCode,
+        final String errorMessage,
+        final Map<String, String> responseHeaders,
+        final CosmosDiagnostics cosmosDiagnostics,
+        final List<ItemBatchOperation<?>> operations) {
+
+        checkNotNull(responseStatus, "expected non-null responseStatus");
+        checkNotNull(responseHeaders, "expected non-null responseHeaders");
+        checkNotNull(operations, "expected non-null operations");
+
+        this.responseStatus = responseStatus;
+        this.subStatusCode = subStatusCode;
+        this.errorMessage = errorMessage;
+        this.responseHeaders = responseHeaders;
+        this.cosmosDiagnostics = cosmosDiagnostics;
+        this.operations = UnmodifiableList.unmodifiableList(operations);
+        this.results = new ArrayList<>();
+    }
+
+    /**
+     * Gets the result of the operation at the provided index in the current {@link TransactionalBatchResponse batch}.
+     * <p>
+     * @param <T> the type parameter.
+     * @param index 0-based index of the operation in the batch whose result needs to be returned.
+     * de-serialized, when present.
+     * @param type class type for which deserialization is needed.
+     *
+     * @return TransactionalBatchOperationResult containing the individual result of operation.
+     */
+    public <T> TransactionalBatchOperationResult<T> getOperationResultAtIndex(
+        final int index,
+        final Class<T> type) {
+
+        checkArgument(index >= 0, "expected non-negative index");
+        checkNotNull(type, "expected non-null type");
+
+        final TransactionalBatchOperationResult<?> result = this.results.get(index);
+        T item = null;
+
+        if (result.getResourceObject() != null) {
+            item = new JsonSerializable(result.getResourceObject()).toObject(type);
+        }
+
+        return new TransactionalBatchOperationResult<T>(result, item);
+    }
+
+    public CosmosDiagnostics getCosmosDiagnostics() {
+        return cosmosDiagnostics;
+    }
+
+    /**
+     * Gets the number of operation results.
+     *
+     * @return the number of operations results in this response.
+     */
+    public int size() {
+        return this.results == null ? 0 : this.results.size();
+    }
+
+    /**
+     * Returns a value indicating whether the batch was successfully processed.
+     *
+     * @return a value indicating whether the batch was successfully processed.
+     */
+    public boolean isSuccessStatusCode() {
+        return this.responseStatus >= 200 && this.responseStatus <= 299;
```

Comment:
`container.createItem()` will always either succeed and returns a CosmosItemResponse or will throw on failure.

however here it seems we are allowing returning a TransactionalBatchResponse even if the whole batch fails. this is not consistent with the above. any reason for the deviation?

shouldn't we follow the same pattern here?

/CC @FabianMeiswinkel  @kirankumarkolli

Reply:
As discussed, user failure will have no exception. So it make sense to keep a function for checking if it succeeded.